### PR TITLE
feat: improved fs uploads performance

### PR DESCRIPTION
**Pull request by grumbach**
- Parallel file uploading with capped concurrency to avoid memory burst
- Parallel chunk uploading with capped concurrency to avoid memory burst
- Local testing reveals this PR improves upload performance of CLI directories upload by **x50** FIFTY!!
- Configurable batching for files and chunks through env vars

### CODE REVIEW

File: autonomi/src/client/fs.rs [hunk 1]
```diff
@@ -8,15 +8,19 @@
...
 use bytes::Bytes;
 use sn_evm::EvmWallet;
 use sn_networking::target_arch::{Duration, SystemTime};
 use std::path::PathBuf;
+use tokio::task::JoinError;
 
 use super::archive::{Archive, ArchiveAddr};
 use super::data::{DataAddr, GetError, PutError};
 
+pub(crate) const FILE_UPLOAD_BATCH_SIZE: usize = 128;
```
Comment (b-zee) [inline autonomi/src/client/fs.rs]: What's the rationale behind 128? Do we need batching at all?

Comment (grumbach) [reply autonomi/src/client/fs.rs]: I tried with unlimited and my memory useage spiked to the moon. 
So it seems you were right about batching, and we probably need some. Local testing revealed that 128 was a sweet spot (at least on my machine), not too high for mem, not too slow.

**Comment by mazzi** [comment]
Good one! What about a couple of tests? To check file integrity and probably we can profile in the future.
What do you think?

File: autonomi/src/client/fs.rs (hunk 1 above)
Comment (b-zee) [reply autonomi/src/client/fs.rs]: I do worry a little about the relative performance (e.g. PC vs Raspberry Pi). Can we relate this value to CPU core count?

https://doc.rust-lang.org/std/thread/fn.available_parallelism.html

Comment (grumbach) [reply autonomi/src/client/fs.rs]: Oooh excellent! Yes!

File: .github/workflows/merge.yml [hunk 2]
```diff
@@ -110,6 +110,10 @@ jobs:
 
       - uses: Swatinem/rust-cache@v2
 
+      - name: Run autonomi tests
+        timeout-minutes: 25
+        run: cargo test --release --package autonomi --lib --features="full,fs"
+
       - name: Run autonomi tests
```
Comment (ermineJose) [inline .github/workflows/merge.yml]: i think the same workflow is repeated twice! **Run autonomi tests**
Maybe some rebase conflicts?

Comment (grumbach) [reply .github/workflows/merge.yml]: Good catch! I'm actually sneaking this back in as it was removed as a side effect of: 
https://github.com/maidsafe/safe_network/pull/2334

Comment (ermineJose) [reply .github/workflows/merge.yml]: oh cool!

**Pull request merged**